src/ms3/bs4_parser.py

```diff
@@ -2004,13 +2004,99 @@ def __getstate__(self):
 ####################### END OF CLASS DEFINITION #######################
 #######################################################################
 
+class Parts(LoggedClass):
+    def __init__(self, soup: bs4.BeautifulSoup, **logger_cfg):
+        super().__init__('Parts', logger_cfg)
+        self.parts_data = {f"part_{i}": part for i, part in enumerate(soup.find_all('Part'), 1)}
+
+    @property
+    def map_staff2part(self) -> dict[list, str]:
+        """Returns the dict in the format {[2, 3]: 'part_1'} for staves 2 and 3 of part 1"""
+        staff2part = {}
+        for key_part, part in self.parts_data.items():
+            staves = [f"staff_{staff['id']}" for staff in part.find_all('Staff')]
+            staff2part.update(dict.fromkeys(staves, key_part))
+        return staff2part
+
+
 class Instrumentation(LoggedClass):
     """Easy way to read and write the instrumentation of a score, that is
     'instrument', 'longName', 'shortName', 'trackName', 'instrumentId'."""
 
     def __init__(self, soup: bs4.BeautifulSoup, **logger_cfg):
         super().__init__('Instrumentation', logger_cfg)
+        self.INSTRUMENT_DEFAULTS = {
+            'Harpsichord': {'instrumentId': 'keyboard.harpsichord',
+                            'longName': 'Harpsichord',
+                            'part_trackName': 'Harpsichord',
+                            'shortName': 'Hch.',
+                            'trackName': 'Harpsichord'},
+            'Piano': {'instrumentId': 'keyboard.piano',
+                      'longName': 'Piano',
+                      'part_trackName': 'Piano',
+                      'shortName': 'Pno.',
+                      'trackName': 'Piano'},
+            'Violoncello': {'instrumentId': 'strings.cello',
+                            'longName': 'Violoncello',
+                            'part_trackName': 'Cello',
+                            'shortName': 'Vc.',
+                            'trackName': 'Violoncello'},
+        }
+        self.part_tracknames = [elem['part_trackName'] for elem in self.INSTRUMENT_DEFAULTS.values()]
         self.soup = soup
+        # 'instrument',
+        self.fields_names = ['longName', 'shortName', 'trackName', 'instrumentId', 'part_trackName']
+        self.parts = Parts(soup)
+        self.text_tags = self.text_tags_fn()  # store references to XML tags
```

Comment:
please rename the variable and the function to something more meaningful

Reply:
Done